stdlib/public/core/Filter.swift

```diff
@@ -140,7 +111,16 @@ public func == <Base : Collection>(
   lhs: LazyFilterIndex<Base>,
   rhs: LazyFilterIndex<Base>
 ) -> Bool {
-  return lhs.base == rhs.base
+  return lhs.baseIndex == rhs.baseIndex
+}
+
+/// Returns `true` iff `lhs` is less than `rhs`.
+@warn_unused_result
+public func < <Base : Collection>(
+  lhs: LazyFilterIndex<Base>,
+  rhs: LazyFilterIndex<Base>
+) -> Bool {
+    return lhs.baseIndex < rhs.baseIndex
 }
```

Comment:
FIXME: we want to forward all Comparable operations to the base implementation.  There could be specializations of other operations that are cheaper than the default implementation.